jest/react-native-env.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+'use strict';
+
+const NodeEnv = require('jest-environment-node').TestEnvironment;
+
+module.exports = class ReactNativeEnv extends NodeEnv {
+  // this is where we'll add our custom logic
```

Comment:
I see an RFC for `exports` is up now, with the expected `react-native` condition: https://github.com/react-native-community/discussions-and-proposals/pull/534/files#diff-40781647168e437e711b426935003e99ce4bddb4c1a5f1ebd61b0218d2b45670R294

Might make sense to override `exportConditions` right away? Or do you want this PR to be 100% compatible, then make changes to the env iteratively?

Reply:
I guess ideally this env would get the option from metro/metro config somehow instead of defining its own

Reply:
> 100% compatible, then make changes to the env iteratively?

yeah I'd rather have this one be compatible, then do things on top. Maybe post a comment in the RFC itself btw?

Reply:
the export one, it will be a fairly important one for a couple reasons so if we have a reminder of this topic in there it's more likely we'll keep it in mind

Reply:
cool, added some comments now 👍